app/services/jwt_token.rb

```diff
@@ -23,9 +23,14 @@ def iat
 
   def decode
     return nil if token.blank?
-    return nil if secret.blank?
-
-    payload = JWT.decode(token, secret.token)
+    if secret.blank?
```

Comment:
is there a better way to detect what type of token this is? Can we get that information out of `decoded_payload`?

Reply:
We should never rely on information in `decoded_payload` as it hasn't been verified (the signature wasn't validated).

Reply:
We verify the signature later, but we need to decide what is the purpose of this token here. We can decode and if that's for SSO, we proceed with verification. If it's not and we just decode like we did before.

I don't like this too much, hence I'm asking for a better way of differentiation. Is it possible to use other header or that breaks jwt conventions? Can we somehow detect the signature is present, meaning this is SSO token? If we detect there's signature, then perform verification (which `JWT.decode` does for us if we pass public key I believe)

Reply:
We can use the Auth-Header. The RFC states to use two parts:
`TYPE SECRET`. The type can be an arbitrary string, e.g. `Basic`. I would suggest using `Bearer` for OIDC and `JWT` for JWT's issued by Foreman.

Reply:
Done.